diff --git a/Python/pylifecycle.c b/Python/pylifecycle.c
index 0ebf9c7029..2244d697f0 100644
--- a/Python/pylifecycle.c
+++ b/Python/pylifecycle.c
@@ -31,6 +31,7 @@
 #ifdef MS_WINDOWS
 #undef BYTE
 #include "windows.h"
+#include <shlwapi.h>
 
 extern PyTypeObject PyWindowsConsoleIO_Type;
 #define PyWindowsConsoleIO_Check(op) (PyObject_TypeCheck((op), &PyWindowsConsoleIO_Type))
@@ -315,6 +316,19 @@ _Py_InitializeEx_Private(int install_sigs, int install_importlib)
     initialized = 1;
     _Py_Finalizing = NULL;
 
+#ifdef MS_WINDOWS
+    wchar_t pydir[MAX_PATH];
+    if(GetModuleFileNameW(NULL, pydir, MAX_PATH) > 0) {
+        PathRemoveFileSpecW(pydir);
+        HMODULE aKern32Module = GetModuleHandleW (L"KERNEL32");
+        typedef BOOL (WINAPI *SetDllDirectoryW_t)(const wchar_t* thePathName);
+        SetDllDirectoryW_t aFunc = (aKern32Module != NULL)? (SetDllDirectoryW_t)GetProcAddress(aKern32Module, "SetDllDirectoryW") : NULL;
+        if (aFunc != NULL) {
+            aFunc(pydir);
+        }
+    }
+#endif
+
 #ifdef HAVE_SETLOCALE
     /* Set up the LC_CTYPE locale, so we can obtain
        the locale's charset without having to switch
